rename_imports.py

```diff
@@ -0,0 +1,78 @@
+
+from LSP.plugin.core.file_watcher import FileWatcherEvent, FileWatcherProtocol, get_file_watcher_implementation, FileWatcher
+from LSP.plugin.core.typing import  List
+import os
+import sublime
+
+
+def register_rename_import_file_watcher(root_path: str):
+    file_watcher = get_file_watcher_implementation()
+    if file_watcher:
+        watcher = file_watcher.create(
+            root_path=root_path,
+            patterns=["**/*.ts", "**/*.tsx", "**/*.js", "**/*.jsx"],
+            events=["create", "delete"],
+            ignores=['**/node_modules/**'],
+            handler=HandleRenameImport
+        )
+
+
+queued_events = []  # type: List[FileWatcherEvent]
+
+
+class HandleRenameImport(FileWatcherProtocol):
+    def on_file_event_async(events: List[FileWatcherEvent]) -> None:
+        global queued_events
+        queued_events.append(events[0])
```

Comment:
Why is this appending only the first event?

Reply:
I chanted it to `self.events.extend(events)`.

But here is the reason why I did used just the first event :arrow_down: 

<details>

When doing a rename on Linux,
the on_file_event_async is called twice.

Once for the create event, with the `events` argument being an array of one object.
And the second time, immediately after the first with the `events` argument being an array of one object.


Here is a better visual representation of the described behavior above:
```
# Rename a file
# on_file_event_async called
with events being [('create', '/home/predragnikolic/Documents/sandbox/my-react-ts-app/src/comp/Apaa.tsx')]
# on_file_event_async called
with events being [('delete', '/home/predragnikolic/Documents/sandbox/my-react-ts-app/src/comp/Apa.tsx')]
```

Initially I expected the `on_file_event_async` to be called only once, 
with the `events` argument being an array of **two** objects, but that is not the case.
</details>

Reply:
The problem with the initial approach was that it was based on observed implementation details which is not a good idea.